plugins/modules/filesize.py

```diff
@@ -0,0 +1,439 @@
+        and so on for C(T), C(P), C(E), C(Z) and C(Y); or alternatively one of
+        C(K), C(k), C(KiB) or C(kiB) (= 1024B); C(M), C(m), C(MiB) or C(miB)
+        (= 1024KiB); C(G), C(g), C(GiB) or C(giB) (= 1024MiB); and so on.
+      - If the multiplicative suffix is not provided, the value is treated as
+        an integer number of blocks of I(blocksize) bytes each (float values
+        will be rounded to the closest integer).
+      - When the I(size) value is equal to the current file size, does nothing.
+      - When the I(size) value is bigger than the current file size, bytes will
+        be appended to the file without truncating it, i.e. without modifying
+        the existing bytes of the file.
+      - When the I(size) value is smaller than the current file size, it will
+        be truncated to the requested value without modifyng bytes before this
+        value.
+      - That means that a file of any arbitrary size can be grown to any other
+        arbitrary size, and then resized down to its initial size without
+        modifying its initial content.
+    type: str
+    required: yes
+  blocksize:
+    description:
+      - Size of blocks, in bytes if not followed by a multiplicative suffix.
+      - The numeric value (before the unit) C(MUST) be an integer (or a C(float)
+        if it equals an integer).
+      - If not set, the size of blocks will be guessed from the OS and commonly
+        result in C(512) or C(4096) bytes, that will be used internally by the
+        module or when I(size) has no unit.
+    type: str
+  source:
+    description:
+      - Device or file that will provide input data to provision the file.
+      - This parameter is ignored when I(sparse=yes).
+    type: path
+    default: /dev/zero
+  force:
+    description:
+      - Whether or not to overwrite the file if it exists, i.e. to truncate it
+        from 0. When C(true), the module is not idempotent.
+      - I(force=yes) and I(sparse=yes) are mutually exclusive.
+    type: bool
+    default: no
+  sparse:
+    description:
+      - Whether or not the file to create should be a sparse file.
+      - This option is effective only on newly created files, or when growing a
+        file, only for the bytes to append.
+      - This option is not supported on OpenBSD and Solaris.
+      - I(force=yes) and I(sparse=yes) are mutually exclusive.
+    type: bool
+    default: no
+
+notes:
+  - This module supports I(check_mode) and I(diff).
+
+requirements:
+  - dd
+
+extends_documentation_fragment: files
+'''
+
+EXAMPLES = r'''
+- name: Create a file of 1G filled with null bytes
+  ansible.posix.filesize:
+    path: /var/bigfile
+    size: 1G
+
+- name: Extend the file to 2G (2*1024^3)
+  ansible.posix.filesize:
+    path: /var/bigfile
+    size: 2G
+
+- name: Reduce the file to 2GB (2*1000^3)
+  ansible.posix.filesize:
+    path: /var/bigfile
+    size: 2GB
+
+- name: Fill a file with random bytes for backing a luks device
```

Comment:
```suggestion
- name: Fill a file with random bytes for backing a LUKS device
```
isn't it an abbreviation? https://en.wikipedia.org/wiki/Linux_Unified_Key_Setup#:~:text=The%20Linux%20Unified%20Key%20Setup,was%20originally%20intended%20for%20Linux.&text=The%20reference%20implementation%20for%20LUKS,as%20the%20disk%20encryption%20backend.